namerd/storage/in-memory/src/main/scala/io/buoyant/namerd/storage/InMemoryDtabStore.scala

```diff
@@ -47,6 +47,19 @@ class InMemoryDtabStore(namespaces: Map[String, Dtab]) extends DtabStore {
     }
   }
 
+  def delete(ns: String): Future[Unit] = {
+    val state = get(ns)
+    dtabStatesMu.synchronized {
+      state.sample match {
+        case Some(_) =>
+          state.update(None)
+          Future.Unit
+        case None =>
+          Future.exception(new DtabNamespaceDoesNotExistException(ns))
+      }
+    }
+  }
```

Comment:
Should we actually remove it from the state cache after updating it?  Won't it continue to be returned from `list()` otherwise?